support-frontend/assets/pages/contributions-landing/components/SetPasswordForm.jsx

```diff
@@ -160,6 +185,43 @@ function SetPasswordForm(props: PropTypes) {
           <span className="component-general-error-message__small-print">{passwordErrorMessage}</span>
         </div> : null
       }
+    </div>);
+}
+
+function renderPasswordResetForm(props: PropTypes) {
+  return (
+    <div>
+      <form onSubmit={onSubmitResetPassword(props)} className={classNameWithModifiers('form', ['contribution'])} noValidate>
+        {renderEmailField(props.email)}
+        <Button
+          appearance="secondary"
+          modifierClasses={['create-account']}
+          aria-label="Create a guardian account"
+          type="submit"
+        >
+          Validate my email address
+        </Button>
+        <Button
+          appearance="greyHollow"
+          aria-label="No thank you"
+          onClick={
+            () => {
+              trackComponentClick('decline-to-reset-password');
+              props.setThankYouPageStage('thankYou');
+            }}
+        >
+          No thank you
+        </Button>
+      </form>
+    </div>);
+}
+
+function SetPasswordForm(props: PropTypes) {
+  return (
+    <div className="set-password__form">
+      {props.guestAccountCreationToken ?
```

Comment:
ah OK, so this gives everyone who is **not** brand new and not signed in (i.e. people with full accounts with passwords, full accounts with social sign-in, full accounts which are validated, full accounts unvalidated, existing guest accounts) the password reset form?

Reply:
This would rely on a backend implementation where we could send all "guests" to the "set password" page and the just-created guests would have a token, and the not-just-created guests would get the reset button. We wouldn't want to send all our traffic to this page, though. Just accurately identified "guests."